src/app/feed/shared/feed-connection.model.spec.ts

```diff
@@ -30,102 +30,248 @@ describe("Service: FeedConnection", () => {
-    let fConnection: FeedConnection = new FeedConnection();
-    fConnection.setAttrs(
-      this.pluginHandle,
-      1
-    );
+  describe("#publish", () => {
+    beforeEach(() => {
+      this.options = jasmine.createSpyObj("options", ["success", "error"]);
+    });
 
-    let jsep: any = { id: 13 };
-    fConnection.handleRemoteJsep(jsep);
+    it("should create a webRTC offer as subscriber", () => {
+      this.connection.publish();
+
+      expect(this.pluginHandle.createOffer).toHaveBeenCalledWith(
+        jasmine.objectContaining({
+          media: {
+            videoRecv: false,
+            audioRecv: false,
+            video: "screen",
+            audioSend: false,
+            data: false
+          },
+          success: jasmine.any(Function),
+          error: jasmine.any(Function)
+        })
+      );
+    });
+
+    it("should create a webRTC offer as main", () => {
+      this.connection.role = "main";
+      this.connection.publish();
+
+      expect(this.pluginHandle.createOffer).toHaveBeenCalledWith(
+        jasmine.objectContaining({
+          media: {
+            videoRecv: false,
+            audioRecv: false,
+            audioSend: true,
+            videoSend: true,
+            data: true
+          },
+          success: jasmine.any(Function),
+          error: jasmine.any(Function)
+        })
+      );
+    });
+
+    it("should call success callback when webRTC offer success", () => {
+      this.pluginHandle.createOffer.and.callFake((options) => {
+        options.success()
+      });
+      this.connection.publish(this.options);
+
+      expect(this.options.success).toHaveBeenCalled();
+    });
+
+    it("should call success callback when webRTC offer success", () => {
+      this.pluginHandle.createOffer.and.callFake((options) => {
+        options.error()
+      });
+      this.connection.publish(this.options);
+
+      expect(this.options.error).toHaveBeenCalled();
+    });
 
-    expect(this.pluginHandle.handleRemoteJsep).toHaveBeenCalledWith(
-      jasmine.objectContaining({
-        jsep: jsep
-      })
-    );
   });
 
-  it("should call pluginHandle.data when call sendData", () => {
-    let fConnection: FeedConnection = new FeedConnection();
-    fConnection.setAttrs(
-      this.pluginHandle,
-      1
-    );
+  describe("#subscribe", () => {
+    beforeEach(() => {
+      this.jsepVal = {};
+    });
+    it("should create a webRTC answer", () => {
+      this.connection.subscribe(this.jsepVal);
+
+      expect(this.pluginHandle.createAnswer).toHaveBeenCalledWith(
+        jasmine.objectContaining({
+          jsep: this.jsepVal,
+          media: {
+            audioSend: false,
+            videoSend: false,
+            data: true
+          },
+          success: jasmine.any(Function),
+          error: jasmine.any(Function)
+        })
+      );
+    });
+
+    it("should send a request start message when webRTC answer success", () => {
```

Comment:
I don't think I'm testing the `pluginHandler`, in these tests the `pluginHandler` it's a fake. In this test I'm faking the `createAnswer` function to test the anonymous callbacks, but this could be a problem if  `pluginHandler` changes his public functions.

De-anonymizing callbacks could be a solution, but IMHO this callback should be private methods and could not be easily testable.